src/main/java/uk/ac/ebi/eva/contigalias/controller/contigalias/ContigAliasController.java

```diff
@@ -144,67 +143,66 @@ public static void linkPagedModelGetChromosomesByAssemblyAccession(
+        } else return new ResponseEntity<>(HttpStatus.REQUESTED_RANGE_NOT_SATISFIABLE);
     }
 
-    // TODO multiple chromosomes have same genbank :/
-    @ApiOperation(value = "Get an assembly using the refseq accession of one of its " +
-            "chromosomes.")
-    @GetMapping(value = "assemblies/chromosome/refseq/{refseq}")
-    public ResponseEntity<PagedModel<EntityModel<AssemblyEntity>>> getAssemblyByChromosomeRefseq(
-            @PathVariable String refseq,
+    @ApiOperation(value = "Get a list of assemblies using the RefSeq accession of one of the chromosomes they have " +
+            "in common.",
+            notes = "Given a chromosome's RefSeq accession, this endpoint will return a list of assemblies that are " +
+                    "associated with a chromosome having the same RefSeq accession as the one provided. This " +
+                    "endpoint returns a list containing one or more assemblies. It also accepts two additional " +
+                    "parameters (page and size) to control pagination of results. If the page number and/or page size" +
+                    " are invalid then an HTTP status code of 416 is returned by this endpoint.")
+    @GetMapping(value = "chromosomes/refseq/{refseq}/assemblies")
+    public ResponseEntity<PagedModel<EntityModel<AssemblyEntity>>> getAssembliesByChromosomeRefseq(
+            @PathVariable @ApiParam(value = "RefSeq accession of the chromosomes.") String refseq,
             @RequestParam(required = false, name = "page") @ApiParam(value = PAGE_NUMBER_DESCRIPTION) Integer pageNumber,
             @RequestParam(required = false, name = "size") @ApiParam(value = PAGE_SIZE_DESCRIPTION) Integer pageSize) {
         if (paramsValidForSingleResponseQuery(pageNumber, pageSize)) {
-            PagedModel<EntityModel<AssemblyEntity>> pagedModel = handler.getAssemblyByChromosomeRefseq(refseq);
+            PagedModel<EntityModel<AssemblyEntity>> pagedModel = handler.getAssembliesByChromosomeRefseq(refseq);
             return createAppropriateResponseEntity(pagedModel);
-        } else return BAD_REQUEST;
+        } else return new ResponseEntity<>(HttpStatus.REQUESTED_RANGE_NOT_SATISFIABLE);
     }
 
-    // TODO Handle issue were result is not unique (Eg: with CM000663.1)
-    @ApiOperation(value = "Get a chromosome using its Genbank accession.",
-            notes = "Given a chromosome's genbank accession this endpoint will return a chromosome that matches that " +
-                    "accession. This endpoint will either return a list containing a single result or an HTTP " +
-                    "Response with error code 404.")
+    @ApiOperation(value = "Get a list of chromosomes using their common GenBank accession.",
+            notes = "Given a chromosome's GenBank accession, this endpoint will return a list of all chromosomes that" +
+                    " match that accession. This endpoint will either return a list of chromosomes.")
```

Comment:
This description and the next one lack the reason why there will be several chromosomes if you just provided 1 chr accession, which is that there will be 1 entry per assembly. This week I have to do a demo of this project to some teams at my workplace and will see how intuitive this is to other people.

Maybe it's easier to understand if we return a list with 1 chr only, and then we provide several hateoas links to link to the assemblies. Arguably, our internal representation is leaking here.

Reply:
So there are multiple ways we can go about this: 
- we can explain in the documentation why we have multiple identical chromosomes that are linked to different assemblies
- we can combine all the chromosomes into a single chromosome that has a list of assemblies embedded in it
- we can return a single chromosome and then provide a HateOAS link to `getAssembliesByChromosomeGenbank`

I think the last option fits in with the way all of our other endpoints function. What do you think?

Reply:
yeah, the last one sounds the best